src/Utils/RequestExpectationComparator.php

```diff
@@ -184,4 +185,40 @@ private function requestFormDataMatchExpectation(ServerRequestInterface $httpReq
 
         return true;
     }
+
+    private function requestJsonMatchesExpectation(ServerRequestInterface $httpRequest, Conditions $expectedRequest): bool
+    {
+        if (!$expectedRequest->hasJsonPath()) {
+            return true;
+        }
+        
+        $this->logger->debug('Checking JSON PATH against expectation');
+        
+        $requestBody = $httpRequest->getBody()->__toString();
+        $requestData = json_decode($requestBody, true);
+        
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            return false;
+        }
+        
+        /** @var JsonPathName $pathName */
+        /** @var JsonPathCondition $jsonCondition */
+        foreach ($expectedRequest->getJsonPath() as $pathName => $jsonCondition) {
+            $path = explode('.', $pathName->asString());
+            $value = $requestData;
+            
+            foreach ($path as $key) {
+                if (!is_array($value) || !isset($value[$key])) {
```

Comment:
[question] From the code I assume we can also use notation like: `users.0.name` in case of indexed arrays. Is that correct? If yes, it would be nice to document it and have a test for it.

Reply:
You are absolutely right about array access notation - `users.0.name` will work for accessing array elements. I will add documentation and test to demonstrate this functionality.